Assets/Tests/EditMode/OriginAwareLaunchPlanningTest.cs

```diff
@@ -0,0 +1,280 @@
+// Tests that launch planning algorithms correctly account for interceptor origins
+// rather than assuming launch from (0,0,0).
+//
+// KEY PRINCIPLE: The intercept position should be where both interceptor and target meet.
+// For successful intercepts, this should be very close to the predicted target position.
+// Origin-aware planning ensures interceptors launch from the correct starting position
+// to reach this intercept point.
+[TestFixture]
+public class OriginAwareLaunchPlanningTest : TestBase {
+  private InterceptorOriginConfig _staticOrigin;
+  private InterceptorOriginConfig _movingOrigin;
+  private MockPredictor _mockPredictor;
+  private MockLaunchAnglePlanner _mockLaunchAnglePlanner;
+
+  [SetUp]
+  public void SetUp() {
+    // Set up test origins
+    _staticOrigin = new InterceptorOriginConfig {
+      id = "Static-Test-Origin", initial_position = new Vector3(1000, 0, 2000),
+      velocity = Vector3.zero, max_interceptors = 10,
+      interceptor_types = new System.Collections.Generic.List<string> { "test.json" }
+    };
+
+    _movingOrigin = new InterceptorOriginConfig {
+      id = "Moving-Test-Origin", initial_position = new Vector3(500, 0, 1000),
+      velocity = new Vector3(0, 0, -10),  // Moving 10 m/s in -Z direction
+      max_interceptors = 5,
+      interceptor_types = new System.Collections.Generic.List<string> { "test.json" }
+    };
+
+    // Set up mock components
+    _mockPredictor = new MockPredictor();
+    _mockLaunchAnglePlanner = new MockLaunchAnglePlanner();
+  }
+
+  // Helper method to create a mock InterceptorOrigin for testing
+  private InterceptorOrigin CreateMockOriginObject(InterceptorOriginConfig config,
+                                                   float currentTime = 0f) {
+    // Create a GameObject to serve as the mock origin object
+    GameObject mockOriginGameObject = new GameObject($"Mock_{config.id}");
+
+    // Calculate position accounting for time and velocity
+    Vector3 currentPosition = config.initial_position + config.velocity * currentTime;
+    mockOriginGameObject.transform.position = currentPosition;
+
+    // Add the InterceptorOrigin component
+    InterceptorOrigin originObject = mockOriginGameObject.AddComponent<InterceptorOrigin>();
+    originObject.SetOriginConfig(config);
+
+    return originObject;
+  }
+
+  [Test]
+  public void TestIterativeLaunchPlanner_WithOrigin_StaticOrigin() {
+    // Test that IterativeLaunchPlanner correctly uses static origin position
+    var planner = new IterativeLaunchPlanner(_mockLaunchAnglePlanner, _mockPredictor);
+
+    // Set up a simple threat moving toward origin
+    Vector3 threatInitialPos = new Vector3(2000, 0, 5000);
+    Vector3 threatVelocity = new Vector3(0, 0, -100);
+    _mockPredictor.SetThreatTrajectory(threatInitialPos, threatVelocity);
+
+    // Set up mock launch angle planner to return reasonable values
+    _mockLaunchAnglePlanner.SetMockResponse(
```

Comment:
instead of commenting, prefer annotating the arguments like:

```suggestion
    _mockLaunchAnglePlanner.SetMockResponse(launchAngle: 45f, timeToIntercept: 10f)
```